packages/admin/cms-admin/src/dam/Table/duplicatedFilenames/DuplicatedFilenamesResolver.tsx

```diff
@@ -0,0 +1,113 @@
+import { useApolloClient } from "@apollo/client";
+import * as React from "react";
+
+import { GQLDamBulkFilenameAlreadyExistsQuery, GQLDamBulkFilenameAlreadyExistsQueryVariables, GQLFilenameResponse } from "../../../graphql.generated";
+import { DuplicatedFilenameDialog } from "./DuplicatedFilenameDialog";
+import { damBulkFilenameAlreadyExistsQuery } from "./DuplicatedFilenamesResolver.gql";
+
+export interface DuplicatedFilenamesResolverApi {
+    checkForDuplicates: (fileData: FileData[]) => Promise<GQLFilenameResponse[]>;
+    resolveDuplicates: (fileData: FileData[], onCancelUpload: () => void, callback: (newFilenames: FileData[]) => unknown) => Promise<void>;
+}
+
+export interface FileData {
+    file: File;
+    folderId?: string;
+}
+
+export const DuplicatedFilenamesResolverContext = React.createContext<DuplicatedFilenamesResolverApi>({
+    checkForDuplicates: () => {
+        throw new Error("DuplicatedFilenamesResolver has to be defined higher up in the tree");
+    },
+    resolveDuplicates: () => {
+        throw new Error("DuplicatedFilenamesResolver has to be defined higher up in the tree");
+    },
+});
+
+export const useDuplicatedFilenamesResolver = (): DuplicatedFilenamesResolverApi => {
+    return React.useContext(DuplicatedFilenamesResolverContext);
+};
+
+export const DuplicatedFilenamesResolver: React.FunctionComponent = ({ children }) => {
+    const client = useApolloClient();
+
+    const [occupiedFilenames, setOccupiedFilenames] = React.useState<FileData[]>([]);
+    const [unoccupiedFilenames, setUnoccupiedFilenames] = React.useState<FileData[]>([]);
+    const [callback, setCallback] = React.useState<(newFilenames: FileData[]) => unknown>();
+
+    const checkForDuplicates = React.useCallback(
+        async (fileData: FileData[]) => {
+            const { data } = await client.query<GQLDamBulkFilenameAlreadyExistsQuery, GQLDamBulkFilenameAlreadyExistsQueryVariables>({
+                query: damBulkFilenameAlreadyExistsQuery,
+                variables: {
+                    filenames: fileData.map((data) => ({
+                        name: data.file.name,
+                        folderId: data.folderId,
+                    })),
+                },
+                fetchPolicy: "network-only",
+            });
+
+            return data.filenamesResponse;
+        },
+        [client],
+    );
+
+    const resolveDuplicates = React.useCallback(
+        async (fileData: FileData[], onCancelUpload: () => void, callback: (newFilenames: FileData[]) => unknown) => {
+            const potentialDuplicates = await checkForDuplicates(fileData);
+
+            const occupiedFilenames: FileData[] = [];
+            const unoccupiedFilenames: FileData[] = [];
+
+            for (const potentialDuplicate of potentialDuplicates) {
+                if (potentialDuplicate.isOccupied) {
+                    occupiedFilenames.push(
+                        fileData.find((data) => {
```

Comment:
Why do you need this `find` statement? Shouldn't it be possible to push `potentialDuplicate`?

Reply:
This code doesn't exist anymore